### PR TITLE
impl. AppendTo and AsSlice

### PR DESCRIPTION
**Pull request by gaissmai**
added AsSlice() and AppendTo() as convenience functions, if you like it, merge it, otherwise drop the PR

thanks for bitset

### CODE REVIEW

**Comment by gaissmai** [comment]
huch, wait a moment, a panic test isn't working as expected

**Comment by gaissmai** [comment]
ready for merge, or drop it if you don't like

File: bitset.go [hunk 1]
```diff
@@ -520,6 +520,44 @@ func (b *BitSet) DeleteAt(i uint) *BitSet {
 	return b
 }
 
+// AppendTo appends all set bits to buf and returns the (maybe extended) buf.
+//
+// See also [BitSet.AsSlice] and [BitSet.NextSetMany].
+func (b *BitSet) AppendTo(buf []uint) []uint {
```
Comment (lemire) [inline bitset.go:526]: Theoretically, we can overflow uint. It would be nice to mention that what happens if someone has a gigantic bitset.

**Comment by lemire** [comment]
@gaissmai Please see my comment.

**Comment by gaissmai** [comment]
I'll will push an update in a few minutes

**Comment by gaissmai** [comment]
I'm not a native speaker, maybe you will polish my new docstring.

**Comment by lemire** [comment]
@gaissmai Let me explain what I meant.

On a 32-bit system, uint maxes out at 2**32 - 1.

Meanwhile, the bitset can represent values between 0 and (2**32 - 1) * 64.

Right?

So `idx<<log2WordSize` might overflow.


It won't overflow in practice. But it is worth saying why. Or, at least, just say 'won't overflow in practice' as a comment.

**Comment by gaissmai** [comment]
hm, yes, but this is not different to NextSet and NetxSetMany

**Comment by lemire** [comment]
> hm, yes, but this is not different to NextSet and NetxSetMany

Correct. Let me merge this and I will add comments.

**Pull request merged**